openapi/eyca.yaml.template

```
@@ -0,0 +1,463 @@
+        "400":
+          description: Bad Request.
+          schema:
+            $ref: "#/definitions/CommonEycaResponse"
+        "401":
+          description: Unauthorized.
+          schema:
+            $ref: "#/definitions/CommonEycaResponse" 
+        "403":
+          description: Forbidden.
+          schema:
+            $ref: "#/definitions/CommonEycaResponse"
+        "404":
+          description: No Account found.
+          schema:
+            $ref: "#/definitions/CommonEycaResponse"
+        "409":
+          description: Conflict.
+          schema:
+            $ref: "#/definitions/CommonEycaResponse"
+        "429":
+          description: Too many requests.
+          schema:
+            $ref: "#/definitions/CommonEycaResponse"
+        "500":
+          description: Service unavailable.
+          schema:
+            $ref: "#/definitions/CommonEycaResponse"
+  "/update_card":
+    get:
+      summary: Perform EYCA Card update
+      operationId: updateCard
+      description: |
+        Perform EYCA Card update and eventually activation
+      parameters:
+        - in: query
+          name: session_id
+          type: string
+          required: true
+          minLength: 1
+        - in: query
+          name: "card_date_expiration"
+          type: string
+          required: true
+        - $ref: "#/parameters/CcdbNumberParam"
+        - $ref: "#/parameters/JsonType"
+      responses:
+        "200":
+          description: Response details.
+          schema:
+            $ref: "#/definitions/CardOperationResponse"
+        "400":
+          description: Bad Request.
+          schema:
+            $ref: "#/definitions/CommonEycaResponse"
+        "401":
+          description: Unauthorized.
+          schema:
+            $ref: "#/definitions/CommonEycaResponse" 
+        "403":
+          description: Forbidden.
+          schema:
+            $ref: "#/definitions/CommonEycaResponse"
+        "404":
+          description: No Account found.
+          schema:
+            $ref: "#/definitions/CommonEycaResponse"
+        "409":
+          description: Conflict.
+          schema:
+            $ref: "#/definitions/CommonEycaResponse"
+        "429":
+          description: Too many requests.
+          schema:
+            $ref: "#/definitions/CommonEycaResponse"
+        "500":
+          description: Service unavailable.
+          schema:
+            $ref: "#/definitions/CommonEycaResponse"
+  
+    
+consumes:
+  - application/json
+produces:
+  - application/json
+
+parameters:
+  JsonType:
+    in: query
+    name: type
+    type: string
+  CcdbNumberParam:
+    in: query
+    name: "ccdb_number"
+    type: string
+    required: true
+  CardDateExpiration:
+    in: query
+    name: "card_date_expiration"
+    type: string
+    required: true
+
+definitions:
+  Timestamp:    
+    $ref: "https://raw.githubusercontent.com/pagopa/io-functions-commons/v21.6.0/openapi/definitions.yaml#/Timestamp"
```

As we moved to templated specs, we can now remove remote references and import straight from local `node_modules`. Benefit: definition versioning is handled alongside the package version
```suggestion
    $ref: "../node_modules/io-functions-commons/openapi/definitions.yaml#/Timestamp"
```

But please note that we are actually using `v17.3.0`, we should consider if we need to upgrade the module or downgrade the definition

---

I know, this was intended to be a first step in that direction. 
I know this is using v17.3.0, but it refers only FiscalCode, Timestamp and ProblemJson, so I thought to leave this PR as small a possible.
I can definitely upgrade fn-commons to v21 ( and remove old italia-ts-commons lib in the process)

---

No, actually I forgot that we cannot upgrade to new fn-commons till we upgrade fp-ts to v2! 
And I cannot use `definitions.yaml` of v17.3.0 because there is no exported openapi in that particular version...